contracts/Vault.sol

```diff
@@ -451,17 +451,17 @@ contract Vault is DBC, Owned, Shares, VaultInterface {
         public
         pre_cond(isSubscribeAllowed)
         pre_cond(isPastZero(incentiveValue))
-        pre_cond(module.pricefeed.isDataValid(MELON_ASSET))
+        pre_cond(module.pricefeed.isValid(MELON_ASSET))
         pre_cond(module.participation.isSubscribeRequestPermitted(
             msg.sender,
             numShares,
             offeredValue
         ))
-        returns(uint256)
+        returns (uint256 id)
```

Comment:
even cleaner would be to substitute `id` w `idCounterOfRequests`
with `idCounterOfRequests++` at beginning of function body

Reply:
then we miss entry `0` don't we?

Reply:
these kinds of counters starting from one was always the case, and used for all similar datatypes throught melon protocol
- they are in line w https://github.com/melonproject/protocol/blob/develop/contracts/Vault.sol#L189
- benefit of: if idCounterOfRequests == 0 then its clear that there haven't been any requests, there is no entry hidden behind id == 0

Reply:
better name for `idCounterOfRequests` would thus be `numberOfRequests`

Reply:
Ok sure.
I remember discussing this and agreeing to make the change to the current, zero-indexed form ([issue](https://github.com/melonproject/protocol/issues/103)), but I'm happy to change them all back if you wish.

I think the one you reference from the vault starts from zero as well, so I would change that one back to one-indexed as well of course.
Opened [this issue](https://github.com/melonproject/protocol/issues/133) and self-assigned.

Reply:
one is `id` the other is `number` of requests
- its fine for `id`s to start from zero
- `number` should start from one!